draft-ietf-masque-connect-udp.md

```diff
@@ -276,34 +276,59 @@ the client (e.g., "connect-udp-version: 2"). Sending this header is RECOMMENDED
 but not required.
 
 
-# Encoding of Proxied UDP Packets {#datagram-encoding}
-
-UDP packets are encoded using HTTP Datagrams {{HTTP-DGRAM}} with the
-UDP_PAYLOAD HTTP Datagram Format Type (see value in {{iana-format-type}}). When
-using the UDP_PAYLOAD HTTP Datagram Format Type, the payload of a UDP packet
-(referred to as "data octets" in {{UDP}}) is sent unmodified in the "HTTP
-Datagram Payload" field of an HTTP Datagram.
-
-In order to use HTTP Datagrams, the client will first decide whether or not it
-will attempt to use HTTP Datagram Contexts and then register its context ID (or
-lack thereof) using the corresponding registration capsule, see {{HTTP-DGRAM}}.
-
-When sending a registration capsule using the "Datagram Format Type" set to
-UDP_PAYLOAD, the "Datagram Format Additional Data" field SHALL be empty.
-Servers MUST NOT register contexts using the UDP_PAYLOAD HTTP Datagram Format
-Type. Clients MUST NOT register more than one context using the UDP_PAYLOAD
-HTTP Datagram Format Type. Endpoints MUST NOT close contexts using the
-UDP_PAYLOAD HTTP Datagram Format Type. If an endpoint detects a violation of
-any of these requirements, it MUST abort the stream.
+# Context Identifiers {#context-id}
+
+This protocol allows future extensions to exchange HTTP Datagrams which carry
+different semantics from UDP payloads. Some of these extensions can augment UDP
+payloads with additional data, while others can exchange data that is completely
+separate from UDP payloads. In order to accomplish this, all HTTP Datagrams
+associated with UDP Proxying request streams start with a context ID, see
+{{format}}.
+
+Context IDs are 62-bit integers (0 to 2<sup>62</sup>-1). The context ID value of
+zero is reserved for UDP payloads, while non-zero values are dynamically
+allocated: non-zero even-numbered context IDs are client-initiated, and
+odd-numbered context IDs are server-initiated. Extensions can use any
+available context ID for their own purposes. Note that, once allocated, any
+context ID can be used by both client and server - only allocation carries
+separate namespaces to avoid requiring synchronization. Additionally, note that
```

Comment:
Is there any constraints on allocation?  As a client, can I perform random allocations from the space?  (yes?) If I allocate 2^62-1 as a server, does that mean that all previous values are taken? (no?) What if a peer attempts to overload my allocation table? (suck it? we'll not be making it easy to consume multiple allocations anyway ...?)

Reply:
Added some text to clarify that random allocations are allowed